public/static/js/projects.js

```diff
@@ -431,21 +423,25 @@ mciModule.controller('ProjectCtrl', function($scope, $window, $http, $location,
   };
 
   $scope.addGithubAlias = function() {
-    if ($scope.github_alias.variant && $scope.github_alias.task) {
-      item = Object.assign({}, $scope.github_alias)
-      item["alias"] = "__github"
-      $scope.github_aliases = $scope.github_aliases.concat([item]);
-      delete $scope.github_alias
-      $scope.invalidPatchDefinitionMessage = "";
-    }
+    if (!$scope.validPatchDefinition($scope.github_alias)) {
```

Comment:
Since the user can't directly change the alias field of the github_alias, I don't think you need to do so much validation here. I don't think you needed to change what was currently there

Reply:
I think you misunderstood existing naming convention. I don't happy with either, but I don't wanted to refactor every piece I don't like. `github_alias` here is not an `alias` property, it's an object, which contain `alias`, `variant`, `task` and `tags`. So validation ensures varant, task or tags are set.